libraries/chain/fork_database.cpp

```diff
@@ -10,6 +10,7 @@
 #include <fc/io/fstream.hpp>
 #include <fc/io/cfile.hpp>
 #include <fstream>
+#include <shared_mutex>
```

Comment:
I think it most likely would be more efficient to make it a `<mutex>` and take exclusive locks everywhere. 
The shared mutex may help if we were doing mostly long reads and an occasional quick mutation here and there. 

But here is is the opposite, the `shared_lock` accesses are very quick so we are not saving much by using read locks, and we pay because the `shared_mutex` is slower than the exclusive mutex.